pkg/libvmi/storage.go

```diff
@@ -278,13 +292,15 @@ func newEmptyDisk(name string, capacity resource.Quantity) v1.Volume {
 	}
 }
 
-func newHostDisk(name, path string, diskType v1.HostDiskType) v1.Volume {
+func newHostDisk(name, path string, diskType v1.HostDiskType, capacity string) v1.Volume {
 	hostDisk := v1.HostDisk{
 		Path: path,
 		Type: diskType,
 	}
+
+	// Set capacity if provided, otherwise default to defaultDiskSize
 	if diskType == v1.HostDiskExistsOrCreate {
-		hostDisk.Capacity = resource.MustParse(defaultDiskSize)
+		hostDisk.Capacity = resource.MustParse(capacity)
```

Comment:
This is wrong: you are setting the capacity only if `diskType == v1.HostDiskExistsOrCreate`.
I'd expect the capacity to be set if it is `!= ""`
Something like:
```
	// Set capacity if provided, otherwise default to defaultDiskSize
	if capacity != "" {
		hostDisk.Capacity = resource.MustParse(capacity)
	}
```

To have compatibility with both `WithHostDisk` and `WithHostDiskAndCapacity`, I'd expect to have something like:
```
func WithHostDisk(diskName, path string, diskType v1.HostDiskType) Option {
	var capacity string
	if diskType == v1.HostDiskExistsOrCreate {
		capacity = defaultDiskSize
	}
	return WithHostDiskAndCapacity(diskName, path, diskType, capacity)
}

func WithHostDiskAndCapacity(diskName, path string, diskType v1.HostDiskType, capacity string) Option {
	return func(vmi *v1.VirtualMachineInstance) {
		addDisk(vmi, newDisk(diskName, v1.DiskBusVirtio))
		addVolume(vmi, newHostDisk(diskName, path, diskType, capacity))
	}
}
```

Reply:
you are right